app/code/Magento/CompareListGraphQl/Model/Resolver/AddItemsToCompareList.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\CompareListGraphQl\Model\Resolver;
+
+use Magento\Catalog\Model\CompareList as ModelCompareList;
+use Magento\Catalog\Model\CompareListFactory;
+use Magento\Catalog\Model\Product\Compare\CompareList;
+use Magento\Catalog\Model\ResourceModel\CompareList as ResourceCompareList;
+use Magento\CompareListGraphQl\Model\Service\AddToCompareListService;
+use Magento\Framework\GraphQl\Config\Element\Field;
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\Store\Api\Data\StoreInterface;
+use Magento\CompareListGraphQl\Model\Service\CompareListService;
+
+/**
+ * Class add products to compare list
+ */
+class AddItemsToCompareList implements ResolverInterface
+{
+    /**
+     * @var CompareListFactory
+     */
+    private $compareListFactory;
+
+    /**
+     * @var ResourceCompareList
+     */
+    private $resourceCompareList;
+
+    /**
+     * @var CompareList
+     */
+    private $compareList;
+
+    /**
+     * @var AddToCompareListService
+     */
+    private $addToCompareListService;
+
+    /**
+     * @var CompareListService
+     */
+    private $compareListService;
+
+    /**
+     * @param CompareListFactory $compareListFactory
+     * @param ResourceCompareList $resourceCompareList
+     * @param CompareList $compareList
+     * @param AddToCompareListService $addToCompareListService
+     * @param CompareListService $compareListService
+     */
+    public function __construct(
+        CompareListFactory $compareListFactory,
+        ResourceCompareList $resourceCompareList,
+        CompareList $compareList,
+        AddToCompareListService $addToCompareListService,
+        CompareListService $compareListService
+    ) {
+        $this->compareListFactory = $compareListFactory;
+        $this->resourceCompareList = $resourceCompareList;
+        $this->compareList = $compareList;
+        $this->addToCompareListService = $addToCompareListService;
+        $this->compareListService = $compareListService;
+    }
+
+    /**
+     * Add items to compare list
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     *
+     * @return Value|mixed|void
+     *
+     * @throws GraphQlInputException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $listId = (int)$args['id'];
```

Comment:
@Usik2203 When clients are using this resolver for the first time, how will they get the listId?

Reply:
@prabhuram93 I was also confused about this case.  
I see that perhaps new mutation will be added for this purpose.
https://github.com/magento/architecture/pull/430/files#diff-184217e17b6d264d529a8363a0bae585R31